packages/components/src/mobile/link-picker/link-picker-results.native.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * External dependencies
+ */
+import { ActivityIndicator, FlatList, View } from 'react-native';
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BottomSheet, BottomSheetConsumer } from '@wordpress/components';
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+
+const PER_PAGE = 20;
+const REQUEST_DEBOUNCE_DELAY = 400;
+const MINIMUM_QUERY_SIZE = 2;
+const meetsThreshold = ( query ) => MINIMUM_QUERY_SIZE <= query.length;
+
+export default function LinkPickerResults( {
+	query,
+	onLinkPicked,
+	directEntry,
+} ) {
+	const [ links, setLinks ] = useState( [ directEntry ] );
+	const [ hasAllSuggestions, setHasAllSuggestions ] = useState( false );
+	const nextPage = useRef( 1 );
+	const pendingRequest = useRef();
+	const clearRequest = () => {
+		pendingRequest.current = null;
+	};
+
+	// a stable debounced function to fetch suggestions and append
+	const { fetchMoreSuggestions } = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		const fetchLinkSuggestions = async ( { search } ) => {
+			if ( meetsThreshold( search ) ) {
+				return await getSettings().__experimentalFetchLinkSuggestions(
+					search,
+					{ page: nextPage.current, type: 'post', perPage: PER_PAGE }
+				);
+			}
+		};
+		const fetchMore = async ( {
+			query: search,
+			links: currentSuggestions,
+		} ) => {
+			// return early if we've already detected the end of data or we are
+			// already awaiting a response
+			if ( hasAllSuggestions || pendingRequest.current ) {
+				return;
+			}
+			const request = fetchLinkSuggestions( { search } );
+			pendingRequest.current = request;
+			const suggestions = await request;
+
+			// only update links for the most recent request
+			if ( suggestions && request === pendingRequest.current ) {
+				// since we don't have the response header, we check if the results
+				// are truncated to determine we've reached the end
+				if ( suggestions.length < PER_PAGE ) {
+					setHasAllSuggestions( true );
+				}
+				setLinks( [ ...currentSuggestions, ...suggestions ] );
+				nextPage.current++;
+			}
+
+			clearRequest();
+		};
+		return {
+			fetchMoreSuggestions: debounce( fetchMore, REQUEST_DEBOUNCE_DELAY ),
+		};
+	}, [] );
```

Comment:
`hasAllSuggestions` should be in the `useSelect` dependency array.

Reply:
Thanks for the comment ecgan :+1: 

It's been a while since I've worked with this code / codebase - but I wonder if simply adding `hasAllSuggestions` to the array might introduce some kind of regression :thinking: . Since `fetchMoreSuggestions` is used in the reset `useEffect`, and is currently expected to be stable, it may need to be added there as well - though I'm not entirely sure off-hand if that can lead to any issues, since the function is debounced (e.g. will this break debouncing in some cases?)

If this is to satisfy a new lint error, and is blocking that work, maybe the simplest solution is adding an ignore - otherwise, this would need further testing in a mobile build to ensure it doesn't introduce regressions, imo.

Reply:
@mkevins , thanks for getting back to me. 🙏 

> I wonder if simply adding `hasAllSuggestions` to the array might introduce some kind of regression 🤔 .

Yes, that's the same concern I have too.

> maybe the simplest solution is adding an ignore

Yeah, I think we can do that. I have just done that in https://github.com/WordPress/gutenberg/pull/49900#discussion_r1171495045.